### DISCUSSION TITLE
Relative links with rewritten paths using next/link

### DISCUSSION THREAD

**Discussion by vevoi** [vercel/next.js#54835] [Help]
### Summary

We have some on page links using href="#my-id" using next/link. The pages these are on have rewrites for teh subpath using nextjs config rewrite:
/subpath/page
/de/de-subpath/page etc

When navigating to a page like this using another next/link (or if a any next/link has loaded?) the final link is correct i.e:

/de/de-subpath/page#my-id

but when navigating straight to this page, the resulting link is not using the rewritten url and is /de/subpath/page#my-id (it still contains the locale).

Is there a way to solve this or do we need to specify the full absolute url in the rewritten format? 

Many thanks!

**Comment by devjiwonchoi** [comment]
Pretty similar to `config.rewrite`, use middleware.

```ts
export function middleware(req) {
  const nextUrl = req.nextUrl
  if (nextUrl.pathname === '/dashboard') {
    if (req.cookies.authToken) {
      return NextResponse.rewrite(new URL('/auth/dashboard', req.url))
    } else {
      return NextResponse.rewrite(new URL('/public/dashboard', req.url))
    }
  }
}
```

### Reference
- [middleware with Link](https://nextjs.org/docs/app/api-reference/components/link#middleware)
